### DISCUSSION TITLE
Missing host to link to!  error when Login is enabled

### DISCUSSION THREAD

**Discussion by jollero** [pglombardo/PasswordPusher#280] [Q&A]
Hi,

I have a heroku deployment. As per advised, I setup the login feature. Upon testing, I keep on nothing this error in the logs.

> ActionView::Template::Error (Missing host to link to! Please provide the :host parameter, set default_url_options[:host], or set :only_path to true):

Any ideas on whats causing it or how to fix it? Thanks a lot in advance!

Regards,
Joseph

**Comment by pglombardo** [comment]
Hi @jollero,

Are you running PasswordPusher ephemeral or with a DB backend?

What version/release of the code are you running?  Git tag, latest master etc...

**Comment by jollero** [comment]
It's the Heroku deployment.

**Comment by pglombardo** [comment]
We've had multiple types and sorts of Heroku deployments through the years.   Even still with Heroku, the code needs to be updated from time to time with a git pull origin and repush to heroku.  Older versions had a bug that causes this which is why I asked.

Assuming you have recent code, then that error message simply means that one of the environment variables is likely not set or at least not set correctly.  I described this [in this issue](https://github.com/pglombardo/PasswordPusher/issues/265#issuecomment-968081491) for the same error.

This error happens when the application is attempting to build fully qualified URLs to send out an email.

Let me know what you find.

**Comment by pglombardo** [comment]
Hi @jollero  - any luck or developments?

`heroku config` shows the proper values now?

**Comment by jollero** [comment]
No luck. Been playing around with config but nothing so far. Will just
probably hold and take our time with this issue for now. Thank you very
much for all you help. You've really built a good and helpful app. :)
